AsyncScrolling: remove duplicated code in ScrollingCoordinator platform implementations

carlosgcampos opened the pull request:
#### 5065b24db42585ced050a5160fbf6c3c362670ba
<pre>
AsyncScrolling: remove duplicated code in ScrollingCoordinator platform implementations
<a href="https://bugs.webkit.org/show_bug.cgi?id=245474">https://bugs.webkit.org/show_bug.cgi?id=245474</a>

Reviewed by Martin Robinson.

Most of the code in Nicosia and Mac implementations is the same but
duplicated in both places. We can move common code to a shared cpp
file.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/scrolling/ScrollingTree.cpp:
(WebCore::ScrollingTree::receivedWheelEvent):
(WebCore::ScrollingTree::deferWheelEventTestCompletionForReason):
(WebCore::ScrollingTree::removeWheelEventTestCompletionDeferralForReason):
* Source/WebCore/page/scrolling/ScrollingTree.h:
(WebCore::ScrollingTree::wheelEventTestMonitor):
(WebCore::ScrollingTree::setWheelEventTestMonitor):
(WebCore::ScrollingTree::setActiveScrollSnapIndices):
(WebCore::ScrollingTree::deferWheelEventTestCompletionForReason): Deleted.
(WebCore::ScrollingTree::removeWheelEventTestCompletionDeferralForReason): Deleted.
(WebCore::ScrollingTree::receivedWheelEvent): Deleted.
* Source/WebCore/page/scrolling/ThreadedScrollingCoordinator.cpp: Copied from Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.mm.
(WebCore::ThreadedScrollingCoordinator::ThreadedScrollingCoordinator):
(WebCore::ThreadedScrollingCoordinator::pageDestroyed):
(WebCore::ThreadedScrollingCoordinator::commitTreeStateIfNeeded):
(WebCore::ThreadedScrollingCoordinator::handleWheelEventForScrolling):
(WebCore::ThreadedScrollingCoordinator::wheelEventWasProcessedByMainThread):
(WebCore::ThreadedScrollingCoordinator::scheduleTreeStateCommit):
(WebCore::ThreadedScrollingCoordinator::didScheduleRenderingUpdate):
(WebCore::ThreadedScrollingCoordinator::willStartRenderingUpdate):
(WebCore::ThreadedScrollingCoordinator::didCompleteRenderingUpdate):
(WebCore::ThreadedScrollingCoordinator::hasNodeWithAnimatedScrollChanged):
(WebCore::ThreadedScrollingCoordinator::startMonitoringWheelEvents):
(WebCore::ThreadedScrollingCoordinator::stopMonitoringWheelEvents):
* Source/WebCore/page/scrolling/ThreadedScrollingCoordinator.h: Copied from Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.h.
* Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.h:
* Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.mm:
(WebCore::ScrollingCoordinatorMac::ScrollingCoordinatorMac):
(WebCore::ScrollingCoordinatorMac::commitTreeStateIfNeeded):
(WebCore::ScrollingCoordinatorMac::pageDestroyed): Deleted.
(WebCore::ScrollingCoordinatorMac::handleWheelEventForScrolling): Deleted.
(WebCore::ScrollingCoordinatorMac::wheelEventWasProcessedByMainThread): Deleted.
(WebCore::ScrollingCoordinatorMac::scheduleTreeStateCommit): Deleted.
(WebCore::ScrollingCoordinatorMac::didScheduleRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorMac::willStartRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorMac::didCompleteRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorMac::hasNodeWithAnimatedScrollChanged): Deleted.
(WebCore::ScrollingCoordinatorMac::startMonitoringWheelEvents): Deleted.
(WebCore::ScrollingCoordinatorMac::stopMonitoringWheelEvents): Deleted.
* Source/WebCore/page/scrolling/mac/ScrollingTreeMac.h:
* Source/WebCore/page/scrolling/mac/ScrollingTreeMac.mm:
(ScrollingTreeMac::setWheelEventTestMonitor): Deleted.
(ScrollingTreeMac::receivedWheelEvent): Deleted.
(ScrollingTreeMac::deferWheelEventTestCompletionForReason): Deleted.
(ScrollingTreeMac::removeWheelEventTestCompletionDeferralForReason): Deleted.
* Source/WebCore/page/scrolling/nicosia/ScrollingCoordinatorNicosia.cpp:
(WebCore::ScrollingCoordinatorNicosia::ScrollingCoordinatorNicosia):
(WebCore::ScrollingCoordinatorNicosia::pageDestroyed): Deleted.
(WebCore::ScrollingCoordinatorNicosia::commitTreeStateIfNeeded): Deleted.
(WebCore::ScrollingCoordinatorNicosia::handleWheelEventForScrolling): Deleted.
(WebCore::ScrollingCoordinatorNicosia::wheelEventWasProcessedByMainThread): Deleted.
(WebCore::ScrollingCoordinatorNicosia::scheduleTreeStateCommit): Deleted.
(WebCore::ScrollingCoordinatorNicosia::willStartRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorNicosia::didCompleteRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorNicosia::hasNodeWithAnimatedScrollChanged): Deleted.
* Source/WebCore/page/scrolling/nicosia/ScrollingCoordinatorNicosia.h:

Canonical link: <a href="https://commits.webkit.org/254952@main">https://commits.webkit.org/254952@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c69f24b97e77ee1e4868b9b717d471fc251a0832

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90812 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35388 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21372 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/100102 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/158609 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/94820 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/33887 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83154 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/96496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96467 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/26988 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77614 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26796 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81750 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81489 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69825 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/34974 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/15524 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32775 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16505 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3455 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36551 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/39456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38476 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35594 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

mrobinson commented on Source/WebCore/page/scrolling/ScrollingTree.cpp:
```diff
@@ -517,6 +517,33 @@ void ScrollingTree::clearLatchedNode()
+{
+    auto monitor = m_wheelEventTestMonitor;
+    if (monitor)
+        monitor->receivedWheelEvent(event);
```
I know this isn't new, but I'm not sure that the temporary variable buys us anything here. I would probably just use the member.

mrobinson commented on Source/WebCore/page/scrolling/ScrollingTree.cpp:
```diff
@@ -517,6 +517,33 @@ void ScrollingTree::clearLatchedNode()
+        return;
+
+    LOG_WITH_STREAM(WheelEventTestMonitor, stream << "    (!) ScrollingTree::deferForReason: Deferring on " << identifier << " for reason " << reason);
+    monitor->deferForReason(identifier, reason);
```
Ditto.

mrobinson commented on Source/WebCore/page/scrolling/ScrollingTree.cpp:
```diff
@@ -517,6 +517,33 @@ void ScrollingTree::clearLatchedNode()
+        return;
+
+    LOG_WITH_STREAM(WheelEventTestMonitor, stream << "    (!) ScrollingTree::removeDeferralForReason: Removing deferral on " << identifier << " for reason " << reason);
+    monitor->removeDeferralForReason(identifier, reason);
```
Ditto.

mrobinson reviewed (approved):
Apologies for the delay. This is a very nice cleanup.

carlosgcampos committed:
AsyncScrolling: remove duplicated code in ScrollingCoordinator platform implementations
https://bugs.webkit.org/show_bug.cgi?id=245474

Reviewed by Martin Robinson.

Most of the code in Nicosia and Mac implementations is the same but
duplicated in both places. We can move common code to a shared cpp
file.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/scrolling/ScrollingTree.cpp:
(WebCore::ScrollingTree::receivedWheelEvent):
(WebCore::ScrollingTree::deferWheelEventTestCompletionForReason):
(WebCore::ScrollingTree::removeWheelEventTestCompletionDeferralForReason):
* Source/WebCore/page/scrolling/ScrollingTree.h:
(WebCore::ScrollingTree::wheelEventTestMonitor):
(WebCore::ScrollingTree::setWheelEventTestMonitor):
(WebCore::ScrollingTree::setActiveScrollSnapIndices):
(WebCore::ScrollingTree::deferWheelEventTestCompletionForReason): Deleted.
(WebCore::ScrollingTree::removeWheelEventTestCompletionDeferralForReason): Deleted.
(WebCore::ScrollingTree::receivedWheelEvent): Deleted.
* Source/WebCore/page/scrolling/ThreadedScrollingCoordinator.cpp: Copied from Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.mm.
(WebCore::ThreadedScrollingCoordinator::ThreadedScrollingCoordinator):
(WebCore::ThreadedScrollingCoordinator::pageDestroyed):
(WebCore::ThreadedScrollingCoordinator::commitTreeStateIfNeeded):
(WebCore::ThreadedScrollingCoordinator::handleWheelEventForScrolling):
(WebCore::ThreadedScrollingCoordinator::wheelEventWasProcessedByMainThread):
(WebCore::ThreadedScrollingCoordinator::scheduleTreeStateCommit):
(WebCore::ThreadedScrollingCoordinator::didScheduleRenderingUpdate):
(WebCore::ThreadedScrollingCoordinator::willStartRenderingUpdate):
(WebCore::ThreadedScrollingCoordinator::didCompleteRenderingUpdate):
(WebCore::ThreadedScrollingCoordinator::hasNodeWithAnimatedScrollChanged):
(WebCore::ThreadedScrollingCoordinator::startMonitoringWheelEvents):
(WebCore::ThreadedScrollingCoordinator::stopMonitoringWheelEvents):
* Source/WebCore/page/scrolling/ThreadedScrollingCoordinator.h: Copied from Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.h.
* Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.h:
* Source/WebCore/page/scrolling/mac/ScrollingCoordinatorMac.mm:
(WebCore::ScrollingCoordinatorMac::ScrollingCoordinatorMac):
(WebCore::ScrollingCoordinatorMac::commitTreeStateIfNeeded):
(WebCore::ScrollingCoordinatorMac::pageDestroyed): Deleted.
(WebCore::ScrollingCoordinatorMac::handleWheelEventForScrolling): Deleted.
(WebCore::ScrollingCoordinatorMac::wheelEventWasProcessedByMainThread): Deleted.
(WebCore::ScrollingCoordinatorMac::scheduleTreeStateCommit): Deleted.
(WebCore::ScrollingCoordinatorMac::didScheduleRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorMac::willStartRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorMac::didCompleteRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorMac::hasNodeWithAnimatedScrollChanged): Deleted.
(WebCore::ScrollingCoordinatorMac::startMonitoringWheelEvents): Deleted.
(WebCore::ScrollingCoordinatorMac::stopMonitoringWheelEvents): Deleted.
* Source/WebCore/page/scrolling/mac/ScrollingTreeMac.h:
* Source/WebCore/page/scrolling/mac/ScrollingTreeMac.mm:
(ScrollingTreeMac::setWheelEventTestMonitor): Deleted.
(ScrollingTreeMac::receivedWheelEvent): Deleted.
(ScrollingTreeMac::deferWheelEventTestCompletionForReason): Deleted.
(ScrollingTreeMac::removeWheelEventTestCompletionDeferralForReason): Deleted.
* Source/WebCore/page/scrolling/nicosia/ScrollingCoordinatorNicosia.cpp:
(WebCore::ScrollingCoordinatorNicosia::ScrollingCoordinatorNicosia):
(WebCore::ScrollingCoordinatorNicosia::pageDestroyed): Deleted.
(WebCore::ScrollingCoordinatorNicosia::commitTreeStateIfNeeded): Deleted.
(WebCore::ScrollingCoordinatorNicosia::handleWheelEventForScrolling): Deleted.
(WebCore::ScrollingCoordinatorNicosia::wheelEventWasProcessedByMainThread): Deleted.
(WebCore::ScrollingCoordinatorNicosia::scheduleTreeStateCommit): Deleted.
(WebCore::ScrollingCoordinatorNicosia::willStartRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorNicosia::didCompleteRenderingUpdate): Deleted.
(WebCore::ScrollingCoordinatorNicosia::hasNodeWithAnimatedScrollChanged): Deleted.
* Source/WebCore/page/scrolling/nicosia/ScrollingCoordinatorNicosia.h:

Canonical link: https://commits.webkit.org/254952@main
